File: HackerRank/Reto-dia-8/main.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

let inputData = '';

process.stdin.on('data', data => inputData += data);

process.stdin.on('end', () => {
    const lines = inputData.split('\n');
    const n = parseInt(lines[0]);
    const phoneBook = new Map();
    
    for (let i = 1; i <= n; i++) {
        const [name, number] = lines[i].split(' ');
        phoneBook.set(name, number);
    }
    
    const output = [];
    for (let i = n + 1; i < lines.length; i++) {
        const name = lines[i].trim();
        if (name === '') continue;
        if (phoneBook.has(name)) {
            output.push(`${name}=${phoneBook.get(name)}`);
        } else {
            output.push('Not found');
        }
    }
    console.log(output.join('\n'));
});

File: HackerRank/Reto-dia-8/main.py
import sys

def main():
    data = sys.stdin.read().split('\n')
    n = int(data[0])
    phone_book = {}
    for i in range(1, n + 1):
        name, number = data[i].split()
        phone_book[name] = number
    
    for line in data[n + 1:]:
        name = line.strip()
        if name == '':
            continue
        if name in phone_book:
            print(f"{name}={phone_book[name]}")
        else:
            print("Not found")

if __name__ == '__main__':
    main()